cli/src/commands/git/push.rs

```diff
@@ -291,10 +300,37 @@ pub fn cmd_git_push(
             }
         }
 
+        for name_revision in &args.named {
+            let (name, target) = create_explicitly_named_bookmarks(ui, &mut tx, name_revision)?;
+            let remote_symbol = name.to_remote_symbol(remote);
```

Comment:
I'm not sure I fully followed your idea for how RefBuf-s, symbols, etc. are supposed to be used, @yuja, and I suspect some part of this could maybe be simpler. Let me know if so.

Reply:
They're just typed strings. `RefNameBuf` and `&RefName` for `String` and `&str` respectively.

Reply:
Well, maybe I used them correctly.

One thing that's a bit confusing to me, BTW, is that I was wondering whether RefNameBuf contains something like `refs/heads/blah`. But also, some code is juggling between strings and RerNameBuf-s, and you might want a different structure to that perhaps.

Reply:
I think that's naming issue. it'll be nice if we can split `RefName` into `BookmarkName` and `TagName`, but we'll need to refactor some bits where these two name types are mixed.